src/c/tb_client/signal.zig

```diff
@@ -274,8 +274,12 @@ pub const Signal = struct {
             .Acquire,
             .Acquire,
         ) orelse {
-            (self.on_signal_fn)(self);
-            return self.wait();
+            const retry = (self.on_signal_fn)(self);
+            self.wait();
+
+            // Wakes up immediately if there are still packets to process.
+            if (retry) self.notify();
+            return;
```

Comment:
I'm not at all familiar with `Signal` — if we are going to wake up immediately, why "wait" at all?

Reply:
If no outgoing requests have finished by the next tick, how are they handled?